### PR TITLE
custom globalExceptionHandler and data validation

### PR DESCRIPTION
**Pull request by kriazantseva**

### CODE REVIEW

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 
 public record CreateBookRequestDto(
-        String title,
-        String author,
-        String isbn,
-        BigDecimal price,
+        @NotNull String title,
+        @NotNull String author,
+        @NotNull @Isbn String isbn,
+        @NotNull @Min(0) BigDecimal price,
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]:
```suggestion
        @NotNull @Positive BigDecimal price,

```

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 import java.math.BigDecimal;
+import mate.academy.bookstoreapp.validation.Isbn;
 
 public record CreateBookRequestDto(
-        String title,
-        String author,
-        String isbn,
-        BigDecimal price,
+        @NotNull String title,
+        @NotNull String author,
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]: Maybe better NoEmpty?

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
+        @NotNull @Isbn String isbn,
+        @NotNull @Min(0) BigDecimal price,
         String description,
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]: You can validate description

File: src/main/java/mate/academy/bookstoreapp/exceptions/CustomGlobalExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP, LocalDateTime.now());
+        body.put(STATUS, HttpStatus.NOT_FOUND);
+        body.put(ERRORS, ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }
+
+    private String getErrorMessage(ObjectError error) {
+        if (error instanceof FieldError) {
+            String fieldName = ((FieldError) error).getField();
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstoreapp/exceptions/CustomGlobalExceptionHandler.java]:
```suggestion
        if (error instanceof FieldError fieldError) {
            String fieldName = fieldError.getField();

```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/bookstoreapp/exceptions/CustomGlobalExceptionHandler.java]: Try to use new features of java

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
 import java.math.BigDecimal;
+import mate.academy.bookstoreapp.validation.Isbn;
 
 public record CreateBookRequestDto(
-        String title,
-        String author,
-        String isbn,
-        BigDecimal price,
-        String description,
+        @NotEmpty String title,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]:
```suggestion
        @NotBlank String title,
```
For String fields

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
+        @NotEmpty String author,
+        @NotNull @Isbn String isbn,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]:
```suggestion
        @NotBlank @Isbn String isbn,

```

File: src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
+import mate.academy.bookstoreapp.validation.Isbn;
 
 public record CreateBookRequestDto(
-        String title,
-        String author,
-        String isbn,
-        BigDecimal price,
-        String description,
+        @NotEmpty String title,
+        @NotEmpty String author,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstoreapp/dto/CreateBookRequestDto.java]:
```suggestion
        @NotBlank String author,

```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's fix some comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**